diff --git a/audio_codec/libamffmpegdec/Makefile b/audio_codec/libamffmpegdec/Makefile
index 69e80d0..e7f1651 100644
--- a/audio_codec/libamffmpegdec/Makefile
+++ b/audio_codec/libamffmpegdec/Makefile
@@ -18,6 +18,10 @@ CC ?=$(CROSS)gcc
 CXX		?= $(CROSS)g++
 LD 		?= $(CROSS)ld
 
+OUTPUT 			= lib$(DW_NAME).so.1.0
+OUTPUT_LN 			= lib$(DW_NAME).so.1
+OUTPUT_SO_LN 			= lib$(DW_NAME).so
+
 DEBUG_CFLAGS     := -c -Wall -shared -fPIC -Wno-format -g -DDEBUG -fexceptions -fnon-call-exceptions -I$(LIBPLAYER_STAGING_DIR)/usr/include
 RELEASE_CFLAGS   := -c -Wall -shared -fPIC -Wno-unknown-pragmas -Wno-format -O3 -fexceptions -fnon-call-exceptions -I$(LIBPLAYER_STAGING_DIR)/usr/include
 DEBUG_LDFLAGS    := -g -shared -fPIC -L$(INSTALL_DIR)  -lavformat -lavcodec -lavutil -lamadec  -soname=$(OUTPUT)
@@ -48,9 +52,6 @@ endef
 
 OBJS 		= $(addsuffix .o,$(basename $(SRC_FILES)))
 DEPENDS 		= $(addsuffix .d,$(basename $(SRC_FILES)))
-OUTPUT 			= lib$(DW_NAME).so.1.0
-OUTPUT_LN 			= lib$(DW_NAME).so.1
-OUTPUT_SO_LN 			= lib$(DW_NAME).so
 
 EXPORT_INCLUDE_FILES := $(addprefix $(SRC_PATH),$(EXPORT_INCLUDE_FILES))
 
diff --git a/audio_codec/libamr/Makefile b/audio_codec/libamr/Makefile
index a5b23a5..97fb04a 100644
--- a/audio_codec/libamr/Makefile
+++ b/audio_codec/libamr/Makefile
@@ -18,6 +18,10 @@ CC ?=$(CROSS)gcc
 CXX	?= $(CROSS)g++
 LD 	?= $(CROSS)ld
 
+OUTPUT 			= lib$(DW_NAME).so.1.0
+OUTPUT_LN 			= lib$(DW_NAME).so.1
+OUTPUT_SO_LN 			= lib$(DW_NAME).so
+
 DEBUG_CFLAGS     := -c -Wall -shared -fPIC -Wno-format -g -DDEBUG -fexceptions -fnon-call-exceptions
 RELEASE_CFLAGS   := -c -Wall -shared -fPIC -Wno-unknown-pragmas -Wno-format -O3 -fexceptions -fnon-call-exceptions
 
@@ -49,9 +53,6 @@ endef
 
 OBJS 		= $(addsuffix .o,$(basename $(SRC_FILES)))
 DEPENDS 		= $(addsuffix .d,$(basename $(SRC_FILES)))
-OUTPUT 			= lib$(DW_NAME).so.1.0
-OUTPUT_LN 			= lib$(DW_NAME).so.1
-OUTPUT_SO_LN 			= lib$(DW_NAME).so
 
 EXPORT_INCLUDE_FILES := $(addprefix $(SRC_PATH),$(EXPORT_INCLUDE_FILES))
 
diff --git a/audio_codec/libape/Makefile b/audio_codec/libape/Makefile
index df45a41..bf64814 100644
--- a/audio_codec/libape/Makefile
+++ b/audio_codec/libape/Makefile
@@ -18,6 +18,10 @@ CC ?=$(CROSS)gcc
 CXX		?= $(CROSS)g++
 LD 		?= $(CROSS)ld
 
+OUTPUT 			= lib$(DW_NAME).so.1.0
+OUTPUT_LN 			= lib$(DW_NAME).so.1
+OUTPUT_SO_LN 			= lib$(DW_NAME).so
+
 DEBUG_CFLAGS     := -c -Wall -shared -fPIC -Wno-format -g -DDEBUG -fexceptions -fnon-call-exceptions
 RELEASE_CFLAGS   := -c -Wall -shared -fPIC -Wno-unknown-pragmas -Wno-format -O3 -fexceptions -fnon-call-exceptions
 
@@ -49,9 +53,6 @@ endef
 
 OBJS 		= $(addsuffix .o,$(basename $(SRC_FILES)))
 DEPENDS 		= $(addsuffix .d,$(basename $(SRC_FILES)))
-OUTPUT 			= lib$(DW_NAME).so.1.0
-OUTPUT_LN 			= lib$(DW_NAME).so.1
-OUTPUT_SO_LN 			= lib$(DW_NAME).so
 
 EXPORT_INCLUDE_FILES := $(addprefix $(SRC_PATH),$(EXPORT_INCLUDE_FILES))
 
diff --git a/audio_codec/libcook/Makefile b/audio_codec/libcook/Makefile
index 2e92fbd..c5a664f 100644
--- a/audio_codec/libcook/Makefile
+++ b/audio_codec/libcook/Makefile
@@ -18,6 +18,10 @@ CC ?=$(CROSS)gcc
 CXX		?= $(CROSS)g++
 LD 		?= $(CROSS)ld
 
+OUTPUT 			= lib$(DW_NAME).so.1.0
+OUTPUT_LN 			= lib$(DW_NAME).so.1
+OUTPUT_SO_LN 			= lib$(DW_NAME).so
+
 DEBUG_CFLAGS     := -c -Wall -shared -fPIC -Wno-format -g -DDEBUG -fexceptions -fnon-call-exceptions
 RELEASE_CFLAGS   := -c -Wall -shared -fPIC -Wno-unknown-pragmas -Wno-format -O3 -fexceptions -fnon-call-exceptions
 
@@ -49,9 +53,6 @@ endef
 
 OBJS 		= $(addsuffix .o,$(basename $(SRC_FILES)))
 DEPENDS 		= $(addsuffix .d,$(basename $(SRC_FILES)))
-OUTPUT 			= lib$(DW_NAME).so.1.0
-OUTPUT_LN 			= lib$(DW_NAME).so.1
-OUTPUT_SO_LN 			= lib$(DW_NAME).so
 
 EXPORT_INCLUDE_FILES := $(addprefix $(SRC_PATH),$(EXPORT_INCLUDE_FILES))
 
diff --git a/audio_codec/libfaad/Makefile b/audio_codec/libfaad/Makefile
index b6d326c..46e219f 100644
--- a/audio_codec/libfaad/Makefile
+++ b/audio_codec/libfaad/Makefile
@@ -20,6 +20,10 @@ CC ?=$(CROSS)gcc
 CXX		?= $(CROSS)g++
 LD 		?= $(CROSS)ld
 
+OUTPUT 			= lib$(DW_NAME).so.1.0
+OUTPUT_LN 			= lib$(DW_NAME).so.1
+OUTPUT_SO_LN 			= lib$(DW_NAME).so
+
 DEBUG_CFLAGS     := -c -Wall -shared -fPIC -Wno-format -g -DDEBUG -fexceptions -fnon-call-exceptions -I$(LIBPLAYER_STAGING_DIR)/usr/include
 RELEASE_CFLAGS   := -c -Wall -shared -fPIC -Wno-unknown-pragmas -Wno-format -O3 -fexceptions -fnon-call-exceptions -I$(LIBPLAYER_STAGING_DIR)/usr/include
 
@@ -56,10 +60,6 @@ OBJS 		= $(addsuffix .o,$(basename $(SRC_FILES)))
 DEPENDS 		= $(addsuffix .d,$(basename $(SRC_FILES)))
 
 
-OUTPUT 			= lib$(DW_NAME).so.1.0
-OUTPUT_LN 			= lib$(DW_NAME).so.1
-OUTPUT_SO_LN 			= lib$(DW_NAME).so
-
 VPATH		= $(SRC_PATH)
 EXPORT_INCLUDE_FILES := $(addprefix $(SRC_PATH),$(EXPORT_INCLUDE_FILES))
 
diff --git a/audio_codec/libflac/Makefile b/audio_codec/libflac/Makefile
index e20e9f8..0447162 100644
--- a/audio_codec/libflac/Makefile
+++ b/audio_codec/libflac/Makefile
@@ -18,6 +18,10 @@ CC ?=$(CROSS)gcc
 CXX		?= $(CROSS)g++
 LD 		?= $(CROSS)ld
 
+OUTPUT 			= lib$(DW_NAME).so.1.0
+OUTPUT_LN 			= lib$(DW_NAME).so.1
+OUTPUT_SO_LN 			= lib$(DW_NAME).so
+
 DEBUG_CFLAGS     := -c -Wall -shared -fPIC -Wno-format -g -DDEBUG -fexceptions -fnon-call-exceptions
 RELEASE_CFLAGS   := -c -Wall -shared -fPIC -Wno-unknown-pragmas -Wno-format -O3 -fexceptions -fnon-call-exceptions
 
@@ -49,9 +53,6 @@ endef
 
 OBJS 		= $(addsuffix .o,$(basename $(SRC_FILES)))
 DEPENDS 		= $(addsuffix .d,$(basename $(SRC_FILES)))
-OUTPUT 			= lib$(DW_NAME).so.1.0
-OUTPUT_LN 			= lib$(DW_NAME).so.1
-OUTPUT_SO_LN 			= lib$(DW_NAME).so
 
 EXPORT_INCLUDE_FILES := $(addprefix $(SRC_PATH),$(EXPORT_INCLUDE_FILES))
 
diff --git a/audio_codec/liblpcm/Makefile b/audio_codec/liblpcm/Makefile
index d0d2ae7..c7a5a52 100644
--- a/audio_codec/liblpcm/Makefile
+++ b/audio_codec/liblpcm/Makefile
@@ -18,6 +18,10 @@ CC ?=$(CROSS)gcc
 CXX		?= $(CROSS)g++
 LD 		?= $(CROSS)ld
 
+OUTPUT 			= lib$(DW_NAME).so.1.0
+OUTPUT_LN 			= lib$(DW_NAME).so.1
+OUTPUT_SO_LN 			= lib$(DW_NAME).so
+
 DEBUG_CFLAGS     := -c -Wall -shared -fPIC -Wno-format -g -DDEBUG -fexceptions -fnon-call-exceptions  -I$(LIBPLAYER_STAGING_DIR)/usr/include
 RELEASE_CFLAGS   := -c -Wall -shared -fPIC -Wno-unknown-pragmas -Wno-format -O3 -fexceptions -fnon-call-exceptions -I$(LIBPLAYER_STAGING_DIR)/usr/include
 
@@ -49,9 +53,6 @@ endef
 
 OBJS 		= $(addsuffix .o,$(basename $(SRC_FILES)))
 DEPENDS 		= $(addsuffix .d,$(basename $(SRC_FILES)))
-OUTPUT 			= lib$(DW_NAME).so.1.0
-OUTPUT_LN 			= lib$(DW_NAME).so.1
-OUTPUT_SO_LN 			= lib$(DW_NAME).so
 
 EXPORT_INCLUDE_FILES := $(addprefix $(SRC_PATH),$(EXPORT_INCLUDE_FILES))
 
diff --git a/audio_codec/libmad/Makefile b/audio_codec/libmad/Makefile
index 94452c8..44318a0 100644
--- a/audio_codec/libmad/Makefile
+++ b/audio_codec/libmad/Makefile
@@ -18,6 +18,10 @@ CC ?=$(CROSS)gcc
 CXX		?= $(CROSS)g++
 LD 		?= $(CROSS)ld
 
+OUTPUT 			= lib$(DW_NAME).so.1.0
+OUTPUT_LN 			= lib$(DW_NAME).so.1
+OUTPUT_SO_LN 			= lib$(DW_NAME).so
+
 DEBUG_CFLAGS     := -c -Wall -shared -fPIC -Wno-format -g -DDEBUG -fexceptions -fnon-call-exceptions -I$(LIBPLAYER_STAGING_DIR)/usr/include
 RELEASE_CFLAGS   := -c -Wall -shared -fPIC -Wno-unknown-pragmas -Wno-format -O3 -fexceptions -fnon-call-exceptions -I$(LIBPLAYER_STAGING_DIR)/usr/include

@@ -49,9 +53,6 @@ endef

 OBJS 		= $(addsuffix .o,$(basename $(SRC_FILES)))
 DEPENDS 		= $(addsuffix .d,$(basename $(SRC_FILES)))
-OUTPUT 			= lib$(DW_NAME).so.1.0
-OUTPUT_LN 			= lib$(DW_NAME).so.1
-OUTPUT_SO_LN 			= lib$(DW_NAME).so

 EXPORT_INCLUDE_FILES := $(addprefix $(SRC_PATH),$(EXPORT_INCLUDE_FILES))

diff --git a/audio_codec/libpcm/Makefile b/audio_codec/libpcm/Makefile
index 15eec05..e0f1b58 100644
--- a/audio_codec/libpcm/Makefile
+++ b/audio_codec/libpcm/Makefile
@@ -18,6 +18,10 @@ CC ?=$(CROSS)gcc
 CXX		?= $(CROSS)g++
 LD 		?= $(CROSS)ld
 
+OUTPUT 			= lib$(DW_NAME).so.1.0
+OUTPUT_LN 			= lib$(DW_NAME).so.1
+OUTPUT_SO_LN 			= lib$(DW_NAME).so
+
 DEBUG_CFLAGS     := -c -Wall -shared -fPIC -Wno-format -g -DDEBUG -fexceptions -fnon-call-exceptions -I$(LIBPLAYER_STAGING_DIR)/usr/include
 RELEASE_CFLAGS   := -c -Wall -shared -fPIC -Wno-unknown-pragmas -Wno-format -O3 -fexceptions -fnon-call-exceptions -I$(LIBPLAYER_STAGING_DIR)/usr/include
 
@@ -49,9 +53,6 @@ endef
 
 OBJS 		= $(addsuffix .o,$(basename $(SRC_FILES)))
 DEPENDS 		= $(addsuffix .d,$(basename $(SRC_FILES)))
-OUTPUT 			= lib$(DW_NAME).so.1.0
-OUTPUT_LN 			= lib$(DW_NAME).so.1
-OUTPUT_SO_LN 			= lib$(DW_NAME).so
 
 EXPORT_INCLUDE_FILES := $(addprefix $(SRC_PATH),$(EXPORT_INCLUDE_FILES))
 
diff --git a/audio_codec/libraac/Makefile b/audio_codec/libraac/Makefile
index 7d4f3e8..f4d6d6b 100644
--- a/audio_codec/libraac/Makefile
+++ b/audio_codec/libraac/Makefile
@@ -18,6 +18,10 @@ CC ?=$(CROSS)gcc
 CXX		?= $(CROSS)g++
 LD 		?= $(CROSS)ld
 
+OUTPUT 			= lib$(DW_NAME).so.1.0
+OUTPUT_LN 			= lib$(DW_NAME).so.1
+OUTPUT_SO_LN 			= lib$(DW_NAME).so
+
 DEBUG_CFLAGS     := -c -Wall -shared -fPIC -Wno-format -g -DDEBUG -fexceptions -fnon-call-exceptions -I$(LIBPLAYER_STAGING_DIR)/usr/include
 RELEASE_CFLAGS   := -c -Wall -shared -fPIC -Wno-unknown-pragmas -Wno-format -O3 -fexceptions -fnon-call-exceptions -I$(LIBPLAYER_STAGING_DIR)/usr/include
 
@@ -49,9 +53,6 @@ endef
 
 OBJS 		= $(addsuffix .o,$(basename $(SRC_FILES)))
 DEPENDS 		= $(addsuffix .d,$(basename $(SRC_FILES)))
-OUTPUT 			= lib$(DW_NAME).so.1.0
-OUTPUT_LN 			= lib$(DW_NAME).so.1
-OUTPUT_SO_LN 			= lib$(DW_NAME).so
 
 EXPORT_INCLUDE_FILES := $(addprefix $(SRC_PATH),$(EXPORT_INCLUDE_FILES))
 
diff --git a/audio_codec/libvorbisdec/Makefile b/audio_codec/libvorbisdec/Makefile
index dfb7892..9975cc4 100644
--- a/audio_codec/libvorbisdec/Makefile
+++ b/audio_codec/libvorbisdec/Makefile
@@ -18,6 +18,10 @@ CC ?=$(CROSS)gcc
 CXX		?= $(CROSS)g++
 LD 		?= $(CROSS)ld
 
+OUTPUT 			= lib$(DW_NAME).so.1.0
+OUTPUT_LN 			= lib$(DW_NAME).so.1
+OUTPUT_SO_LN 			= lib$(DW_NAME).so
+
 DEBUG_CFLAGS     := -c -Wall -shared -fPIC -Wno-format -g -DDEBUG -fexceptions -fnon-call-exceptions
 RELEASE_CFLAGS   := -c -Wall -shared -fPIC -Wno-unknown-pragmas -Wno-format -O3 -fexceptions -fnon-call-exceptions
 
@@ -49,9 +53,6 @@ endef
 
 OBJS 		= $(addsuffix .o,$(basename $(SRC_FILES)))
 DEPENDS 		= $(addsuffix .d,$(basename $(SRC_FILES)))
-OUTPUT 			= lib$(DW_NAME).so.1.0
-OUTPUT_LN 			= lib$(DW_NAME).so.1
-OUTPUT_SO_LN 			= lib$(DW_NAME).so
 
 EXPORT_INCLUDE_FILES := $(addprefix $(SRC_PATH),$(EXPORT_INCLUDE_FILES))
 
diff --git a/audio_codec/wfd_aac_decoder/Makefile b/audio_codec/wfd_aac_decoder/Makefile
index 0db2b1d..23bb358 100644
--- a/audio_codec/wfd_aac_decoder/Makefile
+++ b/audio_codec/wfd_aac_decoder/Makefile
@@ -18,6 +18,10 @@ CC ?=$(CROSS)gcc
 CXX		?= $(CROSS)g++
 LD 		?= $(CROSS)ld
 
+OUTPUT 			= lib$(DW_NAME).so.1.0
+OUTPUT_LN 			= lib$(DW_NAME).so.1
+OUTPUT_SO_LN 			= lib$(DW_NAME).so
+
 DEBUG_CFLAGS     := -c -Wall -shared -fPIC -Wno-format -g -DDEBUG -fexceptions -fnon-call-exceptions
 RELEASE_CFLAGS   := -c -Wall -shared -fPIC -Wno-unknown-pragmas -Wno-format -O3 -fexceptions -fnon-call-exceptions
 
@@ -49,9 +53,6 @@ endef
 
 OBJS 		= $(addsuffix .o,$(basename $(SRC_FILES)))
 DEPENDS 		= $(addsuffix .d,$(basename $(SRC_FILES)))
-OUTPUT 			= lib$(DW_NAME).so.1.0
-OUTPUT_LN 			= lib$(DW_NAME).so.1
-OUTPUT_SO_LN 			= lib$(DW_NAME).so
 
 EXPORT_INCLUDE_FILES := $(addprefix $(SRC_PATH),$(EXPORT_INCLUDE_FILES))
 
