Index: gst-aml-plugins-1.2/video/amlvdec/gstamlvdec.c
===================================================================
--- gst-aml-plugins-1.2.orig/video/amlvdec/gstamlvdec.c
+++ gst-aml-plugins-1.2/video/amlvdec/gstamlvdec.c
@@ -272,7 +272,7 @@ gst_aml_vdec_open(GstVideoDecoder * dec)
     amlvdec->decodetype = FALSE; //for test
     amlvdec->datadump = FALSE; //for debug
     amlvdec->segment.rate = 1.0;
-    amlvdec->tvp = TRUE;
+    amlvdec->tvp = FALSE;
     amlvdec->flushing = FALSE;
 
     return TRUE;
