Index: gst-aml-plugins-1.2/common/amstreaminfo/amlaudioinfo.h
===================================================================
--- gst-aml-plugins-1.2.orig/common/amstreaminfo/amlaudioinfo.h	2017-12-29 10:58:39.122528961 +0800
+++ gst-aml-plugins-1.2/common/amstreaminfo/amlaudioinfo.h	2017-12-29 17:47:12.726988851 +0800
@@ -35,6 +35,7 @@
     unsigned char number_of_raw_data_blocks_in_frame: 2;
 } adts_header_t;
 
+#if 0
 enum CodecID {
     CODEC_ID_NONE,
 
@@ -351,6 +352,7 @@
                                 * stream (only used by libavformat) */
     CODEC_ID_FFMETADATA=0x21000,   ///< Dummy codec for streams containing only metadata information.
 };
+#endif
 
 typedef struct{
     AmlStreamInfo streaminfo;
