From 138c4267ee3c56c62a27c7a5b8600f45d8b50aa1 Mon Sep 17 00:00:00 2001
From: "zhenyu.gao" <zhenyu.gao@amlogic.com>
Date: Fri, 6 May 2022 22:45:54 +0800
Subject: [PATCH] Fix setup.sh for build

Signed-off-by: zhenyu.gao <zhenyu.gao@amlogic.com>
---
 setup.sh | 26 ++++++++++++++++++--------
 1 file changed, 18 insertions(+), 8 deletions(-)

diff --git a/setup.sh b/setup.sh
index 43cd18d..c16f902 100755
--- a/setup.sh
+++ b/setup.sh
@@ -15,6 +15,7 @@ build_dir="build"
 exclude_dir="products docs"
 special_dirs="arch soc boards"
 
+DEFAULT_RTOS_SDK_MANIFEST="$PWD/products/$PRODUCT/rtos_sdk_manifest.xml"
 RTOS_SDK_MANIFEST_FILE="$kernel_BUILD_DIR/rtos_sdk_manifest.xml"
 RTOS_SDK_MANIFEST_OLD_FILE="$kernel_BUILD_DIR/rtos_sdk_manifest_old.xml"
 STAMP="$kernel_BUILD_DIR/.stamp"
@@ -26,14 +27,23 @@ RTOS_SDK_VERSION_FILE="$BUILD_DIR/sdk_ver.h"
 COMPILE_TIME=`date +%F" "%T`
 
 # Check whether the project is a repo
-repo manifest >/dev/null 2>&1
-[ "$?" -ne 0 ] && exit 0
-
-# Generate manifest.xml
-repo manifest > $RTOS_SDK_MANIFEST_FILE
-if [ ! -f $RTOS_SDK_MANIFEST_FILE ]; then
-	echo "Faild to save $RTOS_SDK_MANIFEST_FILE"
-	exit 1
+repo 1234 >/dev/null 2>&1
+if [ "$?" -ne 0 ]; then
+	echo "Non-repo source code"
+	if [ -f $DEFAULT_RTOS_SDK_MANIFEST ]; then
+		echo "Use default manifest: $DEFAULT_RTOS_SDK_MANIFEST"
+		cp -f $DEFAULT_RTOS_SDK_MANIFEST $RTOS_SDK_MANIFEST_FILE
+	else
+		echo "Default manifest.xml not found!"
+		exit 0
+	fi
+else
+	echo "Generate manifest.xml"
+	repo manifest > $RTOS_SDK_MANIFEST_FILE
+	if [ ! -f $RTOS_SDK_MANIFEST_FILE ]; then
+		echo "Faild to save $RTOS_SDK_MANIFEST_FILE"
+		exit 1
+	fi
 fi
 
 if [ -s $RTOS_SDK_MANIFEST_OLD_FILE ] && [ -s $kconfig_file ] && [ $kconfig_file -ot $STAMP ]; then
-- 
2.31.0

