Index: libdrm-2.4.100/meson.build
===================================================================
--- libdrm-2.4.100.orig/meson.build
+++ libdrm-2.4.100/meson.build
@@ -306,7 +306,7 @@ ext_libdrm = declare_dependency(
   include_directories : [inc_root, inc_drm],
 )
 
-install_headers('libsync.h', 'xf86drm.h', 'xf86drmMode.h')
+install_headers('libsync.h', 'xf86drm.h', 'xf86drmMode.h', 'meson_drm.h' )
 install_headers(
   'include/drm/drm.h', 'include/drm/drm_fourcc.h', 'include/drm/drm_mode.h',
   'include/drm/drm_sarea.h', 'include/drm/i915_drm.h',
