Index: src/library.mak
===================================================================
--- src.orig/library.mak
+++ src/library.mak
@@ -19,7 +19,7 @@ $(SUBDIR)$(LIBNAME): $(OBJS)
 	$(AR) $(ARFLAGS) $(AR_O) $^
 	$(RANLIB) $@
 
-install-headers: install-lib$(NAME)-headers install-lib$(NAME)-pkgconfig
+install-headers: install-lib$(NAME)-headers
 
 install-libs-$(CONFIG_STATIC): install-lib$(NAME)-static
 install-libs-$(CONFIG_SHARED): install-lib$(NAME)-shared
