From c1e93b179bceebbd946ba7609fd91ca12abac3c4 Mon Sep 17 00:00:00 2001
From: Pradeep Sriram <pradeep.sriram@amlogic.com>
Date: Wed, 11 Dec 2019 20:15:52 -0800
Subject: [PATCH] add audio hal files

---
 Makefile | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index cc919bc..4a66454 100644
--- a/Makefile
+++ b/Makefile
@@ -9,7 +9,7 @@ LOCAL_PATH = $(shell pwd)
 LDFLAGS += -lstdc++ -lpthread -lz -ldl -lrt -L$(DBUS_LIB_DIR) -ldbus-1
 CFLAGS += -Wall -Wno-unknown-pragmas -Wno-format \
           -O3 -fexceptions -fnon-call-exceptions -D_GNU_SOURCE -DHAVE_AUDIO -I$(DBUS_HEADER_DIR) \
-		  -I$(DBUS_LIB_DIR)/dbus-1.0/include
+		  -I$(DBUS_LIB_DIR)/dbus-1.0/include -I$(STAGING_DIR)/usr/include
 
 ################################################################################
 # libpq.so - src files
@@ -54,6 +54,7 @@ tv_SRCS  = \
 	$(LOCAL_PATH)/libtv/CTv.cpp \
 	$(LOCAL_PATH)/libtv/CTvin.cpp \
 	$(LOCAL_PATH)/libtv/CTvDevicesPollDetect.cpp \
+	$(LOCAL_PATH)/libtv/CTvAudio.cpp \
 	$(NULL)
 
 ################################################################################
-- 
2.7.4

