From 9a511fa30f41f4f1290a57712ea562918dc5beb4 Mon Sep 17 00:00:00 2001
From: Pradeep Sriram <pradeep.sriram@amlogic.com>
Date: Fri, 27 Dec 2019 12:16:23 -0800
Subject: [PATCH] add source switch in test app

---
 test/main_tvtest.cpp | 29 ++++++++++++++++++++++++-----
 1 file changed, 24 insertions(+), 5 deletions(-)

diff --git a/test/main_tvtest.cpp b/test/main_tvtest.cpp
index 5777049..013f0ce 100644
--- a/test/main_tvtest.cpp
+++ b/test/main_tvtest.cpp
@@ -14,6 +14,7 @@
 
 class TvTest: public TvClient::TvClientIObserver {
 public:
+    tv_source_input_t CurrentSource;
     TvTest()
     {
         mpTvClient = TvClient::GetInstance();
@@ -24,7 +25,6 @@ public:
     {
 
     }
-
     void onTvClientEvent(CTvEvent &event)
     {
         int eventType = event.getEventType();
@@ -56,9 +56,9 @@ public:
     int SendCmd(const char *data) {
         LOGD("%s: cmd is %s.\n", __FUNCTION__, data);
         if (strcmp(data, "start") == 0) {
-            mpTvClient->StartTv(SOURCE_HDMI1);
+            mpTvClient->StartTv(CurrentSource);
         } else if (strcmp(data, "stop") == 0) {
-            mpTvClient->StopTv(SOURCE_HDMI1);
+            mpTvClient->StopTv(CurrentSource);
         } else {
             LOGE("invalid cmd!\n");
         }
@@ -101,9 +101,10 @@ int main(int argc, char **argv) {
     char Command[1];
     int run = 1;
     DisplayInit();
+    test->CurrentSource=SOURCE_HDMI1;
 
     LOGD("#### please select cmd####\n");
-    LOGD("#### select s to start####\n");
+    LOGD("#### select 1/2/3 to start####\n");
     LOGD("#### select q to stop####\n");
     LOGD("##########################\n");
     while (run) {
@@ -115,10 +116,28 @@ int main(int argc, char **argv) {
             run = 0;
             break;
           }
-          case 's': {
+          case '1': {
+              test->SendCmd("stop");
+              SetOsdBlankStatus("/sys/class/graphics/fb0/blank", 0);
+              test->CurrentSource=SOURCE_HDMI1;
+              test->SendCmd("start");
+              break;
+          }
+         case '2': {
+              test->SendCmd("stop");
+              SetOsdBlankStatus("/sys/class/graphics/fb0/blank", 0);
+              test->CurrentSource=SOURCE_HDMI2;
+              test->SendCmd("start");
+              break;
+          }
+          case '3': {
+              test->SendCmd("stop");
+              SetOsdBlankStatus("/sys/class/graphics/fb0/blank", 0);
+              test->CurrentSource=SOURCE_HDMI3;
               test->SendCmd("start");
               break;
           }
+ 
           default: {
               test->SendCmd("start");
               break;
-- 
2.7.4

