Index: git/retrace/CMakeLists.txt
===================================================================
--- git.orig/retrace/CMakeLists.txt
+++ git/retrace/CMakeLists.txt
@@ -183,14 +183,14 @@ if (ENABLE_EGL AND ENABLE_WAFFLE)
     )
 
     add_dependencies (eglretrace glproc)
-    include_directories (${Waffle_INCLUDE_DIRS})
+    include_directories (${WAFFLE_INCLUDE_DIRS})
 
     target_link_libraries (eglretrace
         retrace_common
         glretrace_common
         glhelpers
         glproc_egl
-        ${Waffle_LIBRARIES}
+        ${WAFFLE_LIBRARIES}
         ${X11_X11_LIB}
         ${CMAKE_THREAD_LIBS_INIT}
         ${CMAKE_DL_LIBS}
