diff --git a/Makefile b/Makefile
index 10e9b30..ec4a7cd 100644
--- a/Makefile
+++ b/Makefile
@@ -13,8 +13,8 @@ CUTILS_OBJS=src/cutils/hashmap.o \
 	src/cutils/threads.o \
 	src/cutils/strlcpy.o
 
-CFLAGS+=-fPIC -O2 -I./include -I./include/speex -I. -I./src -I./src/ -mfpu=neon -DFIXED_POINT -DRESAMPLE_FORCE_FULL_SINC_TABLE -D_USE_NEON -DEXPORT=
-LDFLAGS+=-llog -ldl -lrt -lpthread -lstdc++
+CFLAGS+=-fPIC -O2 -I./include -I./include/speex -I. -I./src -I./src/ -mfpu=neon -DFIXED_POINT -DRESAMPLE_FORCE_FULL_SINC_TABLE -D_USE_NEON -D_GNU_SOURCE -DEXPORT=
+LDFLAGS+=-ldl -lrt -lpthread -lstdc++
 
 %.o: %.cpp
 	$(CC) -c $(CFLAGS) $(CXXFLAGS) -o $@ $<
@@ -35,28 +35,16 @@ libcutils.so: $(CUTILS_OBJS)
 
 .PHONY: install
 install:
-	install -m 644 -D libamaudioutils.so -t $(STAGING_DIR)/usr/lib
 	install -m 644 -D libamaudioutils.so -t $(TARGET_DIR)/usr/lib
-	install -m 644 -D libcutils.so -t $(STAGING_DIR)/usr/lib
 	install -m 644 -D libcutils.so -t $(TARGET_DIR)/usr/lib
 	for f in $(@D)/include/audio_utils/*.h; do \
-		install -m 644 -D $${f} -t $(STAGING_DIR)/usr/include/audio_utils; \
+		install -m 644 -D $${f} -t $(TARGET_DIR)/usr/include/audio_utils; \
 	done
 	for f in $(@D)/include/audio_utils/spdif/*.h; do \
-		install -m 644 -D $${f} -t $(STAGING_DIR)/usr/include/audio_utils/spdif; \
-	done
-	for f in $(@D)/include/cutils/*.h; do \
-		install -m 644 -D $${f} -t $(STAGING_DIR)/usr/include/cutils; \
-	done
-	for f in $(@D)/include/android/*.h; do \
-		install -m 644 -D $${f} -t $(STAGING_DIR)/usr/include/android; \
+		install -m 644 -D $${f} -t $(TARGET_DIR)/usr/include/audio_utils/spdif; \
 	done
 
 .PHONY: clean
 clean:
-	rm -rf $(STAGING_DIR)/usr/include/audio_utils
-	rm -rf $(STAGING_DIR)/usr/include/cutils
 	rm -f libamaudioutils.so
-	rm -f $(TARGET_DIR)/usr/lib/libamaudioutils.so
-	rm -f $(STAGING_DIR)/usr/lib/libamaudioutils.so
 
