From ac9d43dbc33b0de5a2a977ffdaa075f056ce6622 Mon Sep 17 00:00:00 2001
From: Pradeep Sriram <pradeep.sriram@amlogic.com>
Date: Fri, 6 Dec 2019 14:54:20 -0800
Subject: [PATCH] compile fix

---
 Makefile | 18 ++++++------------
 1 file changed, 6 insertions(+), 12 deletions(-)

diff --git a/Makefile b/Makefile
index 10e9b30..b2c4ade 100644
--- a/Makefile
+++ b/Makefile
@@ -13,8 +13,8 @@ CUTILS_OBJS=src/cutils/hashmap.o \
 	src/cutils/threads.o \
 	src/cutils/strlcpy.o
 
-CFLAGS+=-fPIC -O2 -I./include -I./include/speex -I. -I./src -I./src/ -mfpu=neon -DFIXED_POINT -DRESAMPLE_FORCE_FULL_SINC_TABLE -D_USE_NEON -DEXPORT=
-LDFLAGS+=-llog -ldl -lrt -lpthread -lstdc++
+CFLAGS+=-fPIC -O2 -I./include -I./include/speex -I. -I./src -I./src/ -mfpu=neon -DFIXED_POINT -DRESAMPLE_FORCE_FULL_SINC_TABLE -D_USE_NEON -D_GNU_SOURCE -DEXPORT=
+LDFLAGS+=-ldl -lrt -lpthread -lstdc++
 
 %.o: %.cpp
 	$(CC) -c $(CFLAGS) $(CXXFLAGS) -o $@ $<
@@ -35,28 +35,22 @@ libcutils.so: $(CUTILS_OBJS)
 
 .PHONY: install
 install:
-	install -m 644 -D libamaudioutils.so -t $(STAGING_DIR)/usr/lib
 	install -m 644 -D libamaudioutils.so -t $(TARGET_DIR)/usr/lib
-	install -m 644 -D libcutils.so -t $(STAGING_DIR)/usr/lib
 	install -m 644 -D libcutils.so -t $(TARGET_DIR)/usr/lib
 	for f in $(@D)/include/audio_utils/*.h; do \
-		install -m 644 -D $${f} -t $(STAGING_DIR)/usr/include/audio_utils; \
+		install -m 644 -D $${f} -t $(TARGET_DIR)/usr/include/audio_utils; \
 	done
 	for f in $(@D)/include/audio_utils/spdif/*.h; do \
-		install -m 644 -D $${f} -t $(STAGING_DIR)/usr/include/audio_utils/spdif; \
+		install -m 644 -D $${f} -t $(TARGET_DIR)/usr/include/audio_utils/spdif; \
 	done
 	for f in $(@D)/include/cutils/*.h; do \
-		install -m 644 -D $${f} -t $(STAGING_DIR)/usr/include/cutils; \
+		install -m 644 -D $${f} -t $(TARGET_DIR)/usr/include/cutils; \
 	done
 	for f in $(@D)/include/android/*.h; do \
-		install -m 644 -D $${f} -t $(STAGING_DIR)/usr/include/android; \
+		install -m 644 -D $${f} -t $(TARGET_DIR)/usr/include/android; \
 	done
 
 .PHONY: clean
 clean:
-	rm -rf $(STAGING_DIR)/usr/include/audio_utils
-	rm -rf $(STAGING_DIR)/usr/include/cutils
 	rm -f libamaudioutils.so
-	rm -f $(TARGET_DIR)/usr/lib/libamaudioutils.so
-	rm -f $(STAGING_DIR)/usr/lib/libamaudioutils.so
 
-- 
2.7.4

